src/utils/setup/on-create-page.js

```diff
@@ -50,54 +54,67 @@ const findArticlesFromSlugs = (allArticles, articleSlugs) => {
     return result;
 };
 
-const getLearnPageFilters = async () => {
-    const filters = {
-        languages: {},
-        products: {},
-    };
-
-    // Get possible language and product values from Stitch
-    const languageValues = await stitchClient.callFunction('getValuesByKey', [
-        metadata,
-        'languages',
-    ]);
-    const productValues = await stitchClient.callFunction('getValuesByKey', [
-        metadata,
-        'products',
-    ]);
-
-    // For each language, build an object with its total count, and count for each product
-    for (let i = 0; i < languageValues.length; i++) {
-        const l = languageValues[i];
-        filters.languages[l._id] = {
-            count: l.count,
-            products: {},
-        };
-        const productValuesForLanguage = await stitchClient.callFunction(
-            'getValuesByKey',
-            [metadata, 'products', { languages: l._id }]
-        );
-        productValuesForLanguage.forEach(pl => {
-            filters.languages[l._id].products[pl._id] = pl.count;
-        });
-    }
+const getLearnPageFilters = allArticles => {
+    const languages = {};
+    const products = {};
 
-    // For each product, build an object with its total count, and count for each language
-    for (let i = 0; i < productValues.length; i++) {
-        const p = productValues[i];
-        filters.products[p._id] = {
-            count: p.count,
-            languages: {},
-        };
-        const languageValuesForProduct = await stitchClient.callFunction(
-            'getValuesByKey',
-            [metadata, 'languages', { products: p._id }]
-        );
-        languageValuesForProduct.forEach(lp => {
-            filters.products[p._id].languages[lp._id] = lp.count;
-        });
-    }
-    return filters;
+    allArticles.forEach(article => {
+        const languagesForArticle = article.query_fields.languages;
+        const productsForArticle = article.query_fields.products;
+        // Go through languages for this article and update filter info.
+        // Add a count of how many times this language appears and keep track
+        // of how many itmes we see a product with this language
+        if (languagesForArticle) {
+            languagesForArticle.forEach(language => {
+                const currentLanguage = languages[language];
+                if (currentLanguage) {
+                    currentLanguage.count++;
+                } else {
+                    // Update language reference directly in outer block object
+                    languages[language] = {
+                        count: 1,
+                        products: {},
+                    };
+                }
+                if (productsForArticle) {
+                    // If this article also has products, update those counts as well for this language
+                    const productsForLanguage = languages[language].products;
```

Comment:
@greggb interesting note on these vars. Since we are updating destructively we have to leave off the last key to allow the operation to modify the referenced object and not just the scoped variable :)